compiler/rustc_middle/src/mir/interpret/mod.rs

```diff
@@ -177,7 +177,7 @@ pub enum LitToConstError {
 }
 
 #[derive(Copy, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
-pub struct AllocId(pub u64);
+pub struct AllocId(pub NonZeroU64);
```

Comment:
maybe change the debug impl to print `id - 1` to reduce the test noise in the PR

Reply:
That seems awfully confusing for debugging...